contributors/design-proposals/resource-class.md

```diff
@@ -0,0 +1,363 @@
+# Resource Classes Proposal
+
+  1. [Abstract](#abstract)
+  2. [Motivation](#motivation)
+  3. [Use Cases](#use-cases)
+  4. [Objectives](#objectives)
+  5. [Non Objectives](#non-objectives)
+  6. [Resource Class](#resource-class)
+  7. [API Changes](#api-changes)
+  8. [Scheduler Changes](#sch-changes)
+  9. [Kubelet Changes](#kubelet-changes)
+ 10. [Opaque Integer Resources](#oir)
+ 11. [Future Scope](#future-scope)
+
+_Authors:_
+
+* @vikaschoudhary16 - Vikas Choudhary &lt;vichoudh@redhat.com&gt;
+* @aveshagarwal - Avesh Agarwal &lt;avagarwa@redhat.com&gt;
+
+## Abstract
+In this document we will describe *resource classes* which is a new model to
+represent compute resources in Kubernetes. This document should be seen as a
+successor to [device plugin proposal](https://github.com/kubernetes/community/pull/695/files)
+and has a dependency on the same.
```

Comment:
imho, the proposal is orthogonal to device plugin; device plugin provides more 'primitives' for resource scheduling, while resource claas provides better resource representation.  Resource class is less useful without device plugin so it makes sense to implement it after device plugin, but i'm not quite sure about the dependency here.  Is it a hard dependency?

Reply:
By dependency, i meant the `Device` structure. This proposal needs `Device` to be an API object that is created by device plugins. Also `Device` api object should have a map of device properties.
Now that we a minimum device plugin functionality, i will update this proposal now with the enhancements that are needed in the existing device plugin functionality.